dask_sql/physical/rex/core/call.py

```diff
@@ -99,6 +100,12 @@ def __init__(
         self.false_route = false_route
 
     def apply(self, *operands, **kwargs):
+        if (
+            isinstance(self, CeilFloorOperation)
+            and not isinstance(operands[0], datetime)
+            and len(operands) == 2
+        ):
+            operands = (operands[0].astype("datetime64[ns]"), operands[1])
```

Comment:
Like I mentioned above, think that if we modify the signatures for `timestampceil`/`timestampfloor` on the Rust end we should be able to remove this conditional casting.

That being said, if it needs to stick around, I think it would make more sense to override `apply` in `CeilFloorOperation`, with something like

```python
    def apply(self, *operands, **kwargs):
        if not isinstance(operands[0], datetime):
            operands = (operands[0].astype("datetime64[ns]"), operands[1])
        return super().apply(*operands, **kwargs)
```

Reply:
You're right, changing `variadic` to `exact` removes the need for this check. Thanks!